aep/general/0162/aep.md.j2

```diff
@@ -1,5 +1,372 @@
+**must** be named `revisions`. Resource revisions have paths with the format
+`{resource_path}/revisions/{resource_revision}`. For example:
+```
+publishers/123/books/les-miserables/revisions/c7cfa2a8
+```
+
+### Server-specified aliases
+
+Services **may** reserve specific IDs to be [aliases][alias] (e.g.
+`latest`). These are read-only and managed by the service.
+
+```
+GET /v1/publishers/{publisher_id}/books/{book_id}/revisions/{revision_id}
+```
+
+- If a `latest` ID exists, it **must** represent the most recently created
+revision. The content of `publishers/{publisher}/books/{book}/revisions/latest`
+and `publishers/{publisher}/books/{book}` can differ, as the latest revision may
+be different from the current state of the resource.
+
+### User-specified aliases
+
+APIs **may** provide a mechanism for users to assign an [alias][] ID to an
+existing revision with a custom method "alias":
+
+{% tab proto %}
+
+```proto
+
+rpc AliasBookRevision(TagBookRevisionRequest) returns (Book) {
+  option (google.api.http) = {
+    post: "/v1/{name=publishers/*/books/*/revisions/*}:alias"
+    body: "*"
+  };
+}
+
+message AliasBookRevisionRequest {
+  string path = 1 [
+    (google.api.field_behavior) = REQUIRED,
+    (google.api.resource_reference) = {
+      type: "library.googleapis.com/BookRevision"
+    }];
+
+  // The ID of the revision to alias to, e.g. `CURRENT` or a semantic
+  // version.
+  string alias = 2 [(google.api.field_behavior) = REQUIRED];
+}
+```
+
+{% tab oas %}
+
+```json
+{
+  "paths": {
+    "/v1/publishers/{publisher_id}/books/{book}/revisions/{revision}:alias": {
+      "post": {
+        "requestBody": {
+          "content": {
+            "application/json": {
+              "schema": {
+                "$ref": "#/components/schemas/AliasBookRevisionRequest"
+              }
+            }
+          }
+        },
+        "responses": {
+          "200": {
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/Book"
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  },
+  "components": {
+    "schemas": {
+      "AliasBookRevisionRequest": {
+        "type": "object",
+        "properties": {
+          "path": {
+            "type": "string"
+          },
+          "alias_id": {
+            "type": "string"
+          }
+        },
+        "required": ["path", "alias"]
+      }
+    }
+  }
+}
+```
+
+{% endtabs %}
+
+- The request message **must** have a `path` field:
+  - The field **must** be [annotated as required](/field-behavior-documentation).
+  - The field **must** identify the [resource type](/resource-types) that it
+    references.
+- The request message **must** have a `alias` field:
+  - The field **must** be [annotated as required][aip-203].
+- If the user calls the method with an existing `alias`, the request **must**
```

Comment:
Some tools allow this kind of tagging as described here. Git, notably, errors if you tag a different commit with an existing tag, requiring a `-f` flag to override. Should this behaviors be supported?

This could be introduced as a non-breaking addition by including in the specification an option like `--force-with-lease` which errors if the alias already exists.

Reply:
Good call! I like that. Added a field "overwrite". I think "force" isn't very explicit w.r.t. what behavior it's overwriting. thoughts?

Reply:
resolving for now, but feel free to re-open if you have concerns about the name.